sycl-fusion/jit-compiler/include/Options.h

```diff
@@ -11,67 +11,23 @@
 
 #include "Kernel.h"
 
-#include <memory>
-#include <unordered_map>
-
 namespace jit_compiler {
 
 enum OptionID { VerboseOutput, EnableCaching, TargetDeviceInfo };
 
-class OptionPtrBase {};
-
-class Config {
-
-public:
-  template <typename Opt> void set(typename Opt::ValueType Value) {
-    Opt::set(*this, Value);
-  }
-
-  template <typename Opt> typename Opt::ValueType get() {
-    return Opt::get(*this);
-  }
-
-private:
-  std::unordered_map<OptionID, std::unique_ptr<OptionPtrBase>> OptionValues;
+struct OptionPtrBase {
+  OptionPtrBase(OptionID Id) : Id(Id) {}
```

Comment:
```suggestion
class OptionPtrBase {
 protected:
  explicit OptionPtrBase(OptionID Id) : Id(Id) {}
```
Should only be called from derived class constructors